tests/update_workflow.go

```diff
@@ -2949,17 +2955,20 @@ func (s *FunctionalSuite) TestUpdateWorkflow_ScheduleToStartTimeoutSpeculativeWo
 	s.NoError(err)
 
 	// Now send an update. It will create a speculative WT on normal task queue,
-	// which will time out in 10 seconds and create new normal WT.
+	// which will time out in 5 seconds and create new normal WT.
 	updateResultCh := make(chan *workflowservice.UpdateWorkflowExecutionResponse)
 	go func() {
 		updateResultCh <- s.sendUpdateNoError(tv, "1")
 	}()
 
+	// To make sure that update got to the server before test start to wait for SpeculativeWorkflowTaskScheduleToStartTimeout.
+	runtime.WaitGoRoutineWithFn(s.T(), ((*update.Update)(nil)).WaitLifecycleStage, 1*time.Second)
```

Comment:
No need to change anything but we can use `PollUpdate` for this purpose. (Good to know about the existence of this hack though!)